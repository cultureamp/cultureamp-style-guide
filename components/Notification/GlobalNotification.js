// @flow
import * as React from 'react';
import GenericNotification from './components/GenericNotification';
import type { NotificationType } from './components/GenericNotification';

type Props = {|
  type: NotificationType,
  children: React.Node,
  onHide?: () => void,
|};

const GlobalNotification = (props: Props) => (
  <GenericNotification style="global" persistent={false} {...props} />
);

export default GlobalNotification;
