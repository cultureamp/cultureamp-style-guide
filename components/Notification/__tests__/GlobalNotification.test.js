import React from 'react';
import { mount } from 'enzyme';
import GlobalNotification from '../GlobalNotification';

test('The basic notification renders correctly', () => {
  const notification = mount(
    <GlobalNotification type="positive">
      This is my positive notification
    </GlobalNotification>
  );
  expect(notification).toMatchSnapshot();
});

test('You can change the type', () => {
  const notification = mount(
    <GlobalNotification type="info">
      This is my <em>informational</em> notification
    </GlobalNotification>
  );
  expect(notification).toMatchSnapshot();
});
