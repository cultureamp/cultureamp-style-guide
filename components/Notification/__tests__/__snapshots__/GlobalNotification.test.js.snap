// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The basic notification renders correctly 1`] = `
<GlobalNotification
  type="positive"
>
  <GenericNotification
    persistent={false}
    style="global"
    type="positive"
  >
    <div
      className="notification positive global fading"
    >
      <div
        className="icon"
      >
        <Icon
          icon={null}
          inheritSize={true}
          role="presentation"
        >
          <svg
            aria-hidden={true}
            className="icon inheritSize"
            focusable="false"
            role="presentation"
            viewBox="viewBox"
          >
            <use
              xlinkHref="#id"
            />
          </svg>
        </Icon>
      </div>
      <div
        className="textContainer"
      >
        <span
          className="textWrapper"
        >
          <span
            className="text"
          >
            This is my positive notification
          </span>
        </span>
      </div>
      <span
        className="cancel"
        onClick={[Function]}
      >
        <span
          className="cancelInner"
        >
          <Icon
            icon={null}
            inheritSize={true}
            role="img"
            title="close notification"
          >
            <svg
              aria-labelledby="id-2-title"
              className="icon inheritSize"
              focusable="false"
              role="img"
              viewBox="viewBox"
            >
              <title
                id="id-2-title"
              >
                close notification
              </title>
              <use
                xlinkHref="#id"
              />
            </svg>
          </Icon>
        </span>
      </span>
    </div>
  </GenericNotification>
</GlobalNotification>
`;

exports[`You can change the type 1`] = `
<GlobalNotification
  type="info"
>
  <GenericNotification
    persistent={false}
    style="global"
    type="info"
  >
    <div
      className="notification info global fading"
    >
      <div
        className="icon"
      >
        <Icon
          icon={null}
          inheritSize={true}
          role="presentation"
        >
          <svg
            aria-hidden={true}
            className="icon inheritSize"
            focusable="false"
            role="presentation"
            viewBox="viewBox"
          >
            <use
              xlinkHref="#id"
            />
          </svg>
        </Icon>
      </div>
      <div
        className="textContainer"
      >
        <span
          className="textWrapper"
        >
          <span
            className="text"
          >
            This is my 
            <em>
              informational
            </em>
             notification
          </span>
        </span>
      </div>
      <span
        className="cancel"
        onClick={[Function]}
      >
        <span
          className="cancelInner"
        >
          <Icon
            icon={null}
            inheritSize={true}
            role="img"
            title="close notification"
          >
            <svg
              aria-labelledby="id-4-title"
              className="icon inheritSize"
              focusable="false"
              role="img"
              viewBox="viewBox"
            >
              <title
                id="id-4-title"
              >
                close notification
              </title>
              <use
                xlinkHref="#id"
              />
            </svg>
          </Icon>
        </span>
      </span>
    </div>
  </GenericNotification>
</GlobalNotification>
`;
